urllib3/response.py

```diff
@@ -269,7 +270,18 @@ def read(self, amt=None, decode_content=None, cache_content=False):
             except HTTPException as e:
                 # This includes IncompleteRead.
                 raise ProtocolError('Connection broken: %r' % e, e)
+        except Exception:
+            # The response may not be closed but we're not going to use it anymore
+            # so close it now to ensure that the connection is released back to the pool.
+            if self._original_response and not self._original_response.isclosed():
+                self._original_response.close()
 
+            raise
+        finally:
+            if self._original_response and self._original_response.isclosed():
+                self.release_conn()
+
+        if data:
```

Comment:
How can the execution path make it here without `data` being set?

Won't it just escape via `raise` on L279?